ext/pdo/pdo_stmt.c

```diff
@@ -457,7 +463,14 @@ static inline void fetch_value(pdo_stmt_t *stmt, zval *dest, int colno, int *typ
 	int caller_frees = 0;
 	int type, new_type;
 
-	if (colno < 0 || colno >= stmt->column_count) {
+	if (colno < 0) {
+		zend_value_error("Column index must be greater than or equal to 0");
+		ZVAL_NULL(dest);
+		return;
+	}
+
+	if (colno >= stmt->column_count) {
+		/* TODO Should this be a ValueError? */
```

Comment:
Yes, we did decide to make invalid column/field numbers errors, as you should always know them in advance (unlike row numbers).